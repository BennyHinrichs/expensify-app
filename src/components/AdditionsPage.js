import React from 'react';

const regex = '/^((\d{1,}(\.\d{0,2})?)|(\.\d{1,2})?|(\.)?)$/';

class AdditionsPage extends React.Component {
    render() {
        document.documentElement.scroll(0, 0);
        return (
            <div>
                <div className="page-header">
                    <h1 className="content-container page-header__title">Additions</h1>
                </div>
                <div className="content-container">
                    <p>See the source code: <a href="https://github.com/BennyHinrichs/expensify-app">GitHub repo</a></p>
                    <span>
                        I learned a massive amount from this course, and I couldn't stop myself from continuing on. Here's a (hopefully comprehensive) list of the changes I've made to the project.
                        You can see the original app from Andrew Meade and his insanely good React course <a href="http://expensify.mead.io/">here</a>.
                        If you haven't already, try adding some data points and checking out the features (graphing, editing, filtering).
                    </span>
                    <ul className="additions-list">
                        <li className="additions-list-section">Login Page</li>
                            <ul className="additions-list__sub">
                                <li>Background</li>
                                    <ul className="additions-list__sub__sub">
                                        <li className="additions-list-item">Got a different image (I know, be impressed).</li>
                                        <li className="additions-list-item">Added a vignette via box-shadow.</li>
                                        <li className="additions-list-item">Changed the opacity, added a background color over it with the blend mode color dodge.</li>
                                    </ul>
                                <li>Logo</li>
                                    <ul className="additions-list__sub__sub">
                                        <li className="additions-list-item">Made an actual logo with a fun little stylized S.</li>
                                        <li className="additions-list-item">There are two different logo images: one for desktop and one for mobile.  The desktop one has the subtitle built into the image. The mobile one renders text instead for the subtitle.</li>
                                        <li className="additions-list-item">This seems a good place to mention the favicon and loading animation. I used the $ from the logo to make my own icons.</li>
                                    </ul>
                                <li>Button</li>
                                    <ul className="additions-list__sub__sub">
                                        <li className="additions-list-item">Updated color to match scheme.</li>
                                        <li className="additions-list-item">Added the Google icon.</li>
                                    </ul>          
                            </ul>
                        <li className="additions-list-section">Dashboard</li>
                            <ul className="additions-list__sub">
                                <li>Header</li>
                                    <ul className="additions-list__sub__sub">
                                        <li className="additions-list-item">And there it is again, the logo. This time I used CSS to recolor it. I also made it the background image of a div. Had to use a trick where you set the height to 0 and padding to 10-15% to make it work.</li>
                                        <li className="additions-list-item">The logout text now has a hover animation. I breifly flirted with making all buttons turn yellow on hover, but I decided it looked tacky.</li>
                                        <li className="additions-list-item">The summary is now centered. Some margin adjustment.</li>
                                    </ul>
                                <li>Expense List</li>
                                    <ul className="additions-list__sub__sub">
                                        <li className="additions-list-item">Note is now visible on hover; right aligned for mobile.</li>
                                        <li className="additions-list-item">The style truncates both the description and note with ellipses.</li>
                                    </ul>
                            </ul>
                        <li className="additions-list-section">Graph</li>
                            <ul className="additions-list__sub">
                                <li>Overview</li>
                                    <ul className="additions-list__sub__sub">
                                        <li className="additions-list-item">So this is the brand new component that I did completely solo! I knew going into it that it would be difficult, but it was a good challenge.</li>
                                        <li className="additions-list-item">Some things (like handling fringe cases) took way longer than I anticipated. Others (hooking the graph up to render on filter change) took much shorter than I would have guessed. That's programming though, I suppose.</li>
                                        <li className="additions-list-item">The first thing you'll probably notice is the new "Show Graph" button right-aligned. Try clicking it when you have no data and when you have several data points.</li>
                                    </ul>
                                <li>Sort By</li>
                                    <ul className="additions-list__sub__sub">
                                        <li className="additions-list-item">It quickly became apparent I needed two different types of graphs for the two different sort methods. That meant processing the data accordingly. Can't tell you how long it took to figure out the proper datastructure to work with Chart.js haha.</li>
                                        <li className="additions-list-item">After much war, I got the tooltips working. Got everything working, really, for the scatter. Then when I tried to implement the line graph for sortBy amount, war was waged once more.</li>
                                        <li className="additions-list-item">Nothing would get in the right order. First it was the labels, then the data, then the titles. It always seemed like just when I had it figured out with the data and labels, I'd look and realize yet another thing was wrong.</li>
                                    </ul>
                                <li>Labeling</li>
                                    <ul className="additions-list__sub__sub">
                                        <li className="additions-list-item">To get my data into proper order and format, I had to use Chart.js's callback functions. What frazzled my mind was that I had a callback for the labels and one for the titles, both looking like <span className="additions-list-item--code">(tooltipItem, data) => {'{}'}</span>. To access tooltipItem in title, you had to put an index on it, but not so in label! Who even comes up with this stuff?</li>
                                        <li className="additions-list-item">An annoying aspect of the callbacks is that I needed them, in addition to other parts of my code, to run certain ways based off a conditon. I wanted to do all the conditional stuff in one place, but couldn't figure out a way and still get my data where it needed to be, so I had to duplicate that conditional code.</li>
                                        <li className="additions-list-item">The spread operator saved my life at least five different times in this component alone.</li>
                                        <li className="additions-list-item">I tried the fringe case of really long descriptions and ended up making a truncator function that cuts off incoming strings and appends ellipses.</li>
                                        <li className="additions-list-item">And then the labels would all drop except the first and last if I had more than several points. I eventually found that deactivating <span className="additions-list-item--code">autoSkip</span> takes care of that!</li>
                                    </ul>
                                <li>Fringe Cases</li>
                                    <ul className="additions-list__sub__sub">
                                        <li className="additions-list-item">First we have the case of no data. I decided to render no graph in this case and to just throw up a message instead.</li>
                                        <li className="additions-list-item">Next is the case with one point. This took an unngrokkable amount of time to wrangle down. The graph was only showing a very unattractive point on the y-axis. I needed it centered. What I had to do was add null points a day before and after the single point to make it look nice.</li>
                                        <li className="additions-list-item">But that's not the end. You see, there's also the case of multiple points that have the same values, thus apprearing to be a single point. Now we have to compare and see if the points are the same, then add time-spaced null values to the ends. I put a tolerance of 12 hours on it.</li>
                                        <li className="additions-list-item">Once I got all the padding worked out, of course the labels and tooltips were all out of wack, but only for specific cases. I went in and tracked down the issues with each case, so the graph should now handle any (not gigantic) set of data.</li>
                                    </ul>
                            </ul>
                        <li className="additions-list-section">Expense Form</li>
                            <ul className="additions-list__sub">
                                <li>Regular Expression</li>
                                    <ul className="additions-list__sub__sub">
                                        <li className="additions-list-item">I added support for the amount regex to accept a period alone, because sometimes you want to type a value like .95. That's the third section of the regex. The second section accounts for inputs after you've typed the decimal and allows one to two digits. So you can enter things like .6 now.</li>
                                        <li className="additions-list-item--code additions-list-item">{regex}</li>
                                        <li className="additions-list-item">However, this now opens it up to the possibility of the user entering just a decimal point in the field. Never fear, the app won't allow it! We just pop up a cheeky message telling them to get their head in the game and we're golden.</li>
                                    </ul>
                                <li>Delete</li>
                                    <ul className="additions-list__sub__sub">
                                        <li className="additions-list-item">There is now a confirmation message that pops up when you press delete.</li>
                                        <li className="additions-list-item">After three seconds, the message fades away and you have to press delete again (twice) to remove the expense. Want to prevent accidents!</li>
                                    </ul>
                            </ul>
                            <li className="additions-list-section">TODO</li>
                            <ul className="additions-list__sub">
                                <li>Graph Links</li>
                                    <ul className="additions-list__sub__sub">
                                        <li className="additions-list-item">I might just say I'm done now and move onto another project, but one potential tweak that keeps calling my name is to add links to the graph points that take you to the edit page.</li>
                                        <li className="additions-list-item">Scope creep can be a problem with any project. I'm afraid that if I try and fulfill this TODO list, I'll have another one by the time I'm done.</li>
                                    </ul>
                                <li>Categories and Pie Chart</li>
                                    <ul className="additions-list__sub__sub">
                                        <li className="additions-list-item">If I put in categories that you can tag expenses in, then I could add in a pie chart feature that shows how much each category consumes.</li>
                                        <li className="additions-list-item">If I implemented this and the graph links, I'd have to deactivate graph links for this. That or give each section the same function as the "Show List" button, and have it set the filter as well.</li>
                                    </ul>
                                <li>Browser Detection</li>
                                <ul className="additions-list__sub__sub">
                                    <li className="additions-list-item">Google OAuth doesn't work on iOS native browsers, (e.g. FB Messenger or Snapchat), so I have to detect and pop-up a message.</li>
                                    <li className="additions-list-item">I don't have an iPhone (the main culprit; Android works), which makes it hard to test.</li>
                                </ul>
                                <li>Forecaster</li>
                                    <ul className="additions-list__sub__sub">
                                        <li className="additions-list-item">This is an idea for an additional component. It takes the data that you've filtered and estimates your dollar value expenditures for the next {'{week | month | year}'}.</li>
                                        <li className="additions-list-item">I'd have to figure out exactly what supervised machine learning technique would be best. Maybe something like SVM?</li>
                                    </ul>
                            </ul>
                    </ul>
                </div>
            </div>
        )
    }
}

export default AdditionsPage;