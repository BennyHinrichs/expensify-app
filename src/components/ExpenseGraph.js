import React from 'react';
import { connect } from 'react-redux';
import { Line } from 'react-chartjs-2';
import moment from 'moment';

/* *******TODO******* */
/* ****************** */
/*  1. Animate on filter change?
    2. Hyperlink graph points? https://stackoverflow.com/questions/45980436/chart-js-link-to-other-page-when-click-on-specific-section-in-chart
    
/* ****************** */

const data = {
    datasets: [
    {
        borderColor: '#ffb931',
        fill: false,
        label: 'Expenses',
        pointBorderColor: '#364051',
        pointBackgroundColor: '#fff',
        pointBorderWidth: 1,
        pointHoverRadius: 5,
        pointHoverBackgroundColor: '#364051',
        pointHoverBorderColor: 'rgba(220,220,220,1)',
        pointHoverBorderWidth: 2,
        pointRadius: 3,
        pointHitRadius: 10,
        showLine: true,
    }
    ]
};

const timeAxis = [{
    type: 'time',
    distribution: 'linear',
    time: {
        displayFormats: {
            day: 'D MMM YY'
        },
        unit: 'day'
    }
}]

const amountAxis = [{
    ticks: {
        callback: (value, index, values) => {
            return '$' + value.toFixed(2);
        }
    }
}]

const options  = {
    legend: {
        display: false,
    },
    scales: {
        xAxes: [{
            ticks: {
                autoSkip: false,
            }
        }],
        yAxes: amountAxis
    }
};

const truncator = (str) => {
    const length = 25;
    str = str.length > length ? str.substring(0, length) + '...' : str;
    return str;
}

export class ExpenseGraph extends React.Component {

    // For some insane reason, if I declare this in expenseParser(), the only place it's used,
    // it breaks the graph when sortBy is date (changes the x-axis from time to description labels).
    isSortByDate = () => this.props.filters.sortBy === 'date' ? true : false;

    // Yeah, I could have put this in expenseParser, but that boy was getting big
    strip = (expenses) => {
        let xy = [];
        let title = [];
        for (const i in expenses) {
                xy.push({x: expenses[i].createdAt, y: (expenses[i].amount / 100).toFixed(2)});
                title.push(truncator(expenses[i].description));
            }
        return [xy, title];
    }

    // Wears the pants in this class. Basically takes the data from props and grooms it to fit the
    // chartjs datastructe requirements. Handles whether the user has sorted by date or amount.
    expenseParser = (props) => {
        const arr = this.strip(props.expenses);
        const tooltips = {
            displayColors: false,
            callbacks: { 
                title: (tooltipItem, data) => {
                    // if sortBy is 'amount', reverse title array
                    const title = this.isSortByDate() ? 
                        arr[1][tooltipItem[0].index] :
                        arr[1][arr[1].length - 1 - tooltipItem[0].index]
                    return title;
                },
                label: (tooltipItem, data) => {
                    const createdAt = moment(arr[0][tooltipItem.index].x).format('D MMM YY')
                    const amount = arr[0][tooltipItem.index].y
                    return '$' + amount + ' - ' + createdAt;
                }
            }
        }
        // if sortBy is 'amount', set labels and reverse expense array
        let labels = {};
        if (!this.isSortByDate()) {
            let labelArray = [];
            for (const i in arr[1]) labelArray.push(arr[1][i]);
            labels = { labels: labelArray.reverse() };
            arr[0] = arr[0].reverse();
        }
        // if sortBy is 'date', set axis scales
        const scales = this.isSortByDate() && { scales: { xAxes: timeAxis, yAxes: amountAxis }};
        // apply padding for single value datasets/overlapping datasets that appear as a single value
        const whether = (array) => {
            const len = array.length;
            let isSame = false;
            for (let i = 1; i < len; i++) {
                // I put the arbitrary value of a 12 hour difference here.
                if (array[0].x - array[i].x >= 43200000) {
                    isSame = false;
                    break;
                } else {
                    isSame = true;
                }
            }
            return isSame;
        }
        if (arr[0].length === 1 || whether(arr[0])) {
            const arrCopy = Object.assign({}, arr);
            arr[0].unshift({x: arrCopy[0][0].x - 86400000, y: null});
            arr[0].push({x: arrCopy[0][0].x + 2 * 86400000, y: null});
        }
        // combine all the data into the very specific chartjs data structure
        const dataCombine = {...labels, datasets: [{...data.datasets[0], data: arr[0]}]};
        return [
            dataCombine, 
            {...options, ...scales, tooltips: tooltips}
        ];
    }

    render() {
        const parsedData = this.expenseParser(this.props);
        const Chart = <Line data={ parsedData[0] } options={parsedData[1] } redraw />;
        return (
            <div>
                {(() => {
                    /*This addresses the cases when the user tries to graph 0 or 1 expenses*/
                    switch (this.props.expenses.length){
                        case 0:
                            return (
                                <p className="list-item list-item--graph-message">
                                    You honestly want me to graph nothing? I'm a computer, not a philosopher.
                                </p>
                            );
                        case 1:
                            return (
                                <div>
                                    <p className="list-item list-item--graph-message">
                                        A graph of a single point. Fancy.
                                    </p>
                                    {Chart}
                                </div>
                            );
                        default:
                            return Chart;
                    }
                })()}
            </div>
        );
    }
}

const mapStateToProps = (state) => ({ filters: state.filters });
export default connect(mapStateToProps)(ExpenseGraph);