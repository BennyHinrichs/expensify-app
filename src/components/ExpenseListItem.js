import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';

export default ({ id, description, amount, createdAt, note }) => (
    <Link className="list-item" to={`/edit/${id}`}>
        <div className="list-item__container">
            <div className="list-item__left">
                <h3 className="list-item__title">{description}</h3>
                <span className="list-item__sub-title">{moment(createdAt).format('D MMM YYYY')}</span>
            </div>
            <span className="list-item__note">{note}</span>    
        </div>
        <h3 className="list-item__data">${(amount / 100).toFixed(2)}</h3>
    </Link>
);

// This is how you do it without destructuring
// Note: in ExpenseList.js you have to pass in expense={expense} instead of {...expense}
//
// export default (props) => (
//     <div>
//         <h3>{props.expense.description}</h3>
//         <p>${(props.expense.amount / 100).toFixed(2)}</p>
//         <p>Created At: {props.expense.createdAt}</p>
//     </div>
// );