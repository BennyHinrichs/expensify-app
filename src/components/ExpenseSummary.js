import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

export const ExpenseSummary = (props) => {
    const expenseCount = props.expenses.length;
    const expensesTotal = selectExpensesTotal(props.expenses);

    // expenseText merely accounts for whether it should say '1 expense' or 'n expenses'
    const expenseText = expenseCount === 1 ? 'expense' : 'expenses';

    return (
        <div className="page-header">
            <div className="content-container">
                <h1 className="page-header__title page-header__title--summary">
                    Viewing <span>{expenseCount}</span> {expenseText} totalling <span>${(expensesTotal / 100).toFixed(2)}</span>
                </h1>
                <div className="page-header__actions">
                    <Link className="button" to="/create">Add Expense</Link>
                </div>
            </div>
        </div>
    );
}

// have to use selectExpenses same as ExpenseList to get the proper length (and ergo total)
const mapStateToProps = (state) => ({ expenses: selectExpenses(state.expenses, state.filters )});

export default connect(mapStateToProps)(ExpenseSummary);