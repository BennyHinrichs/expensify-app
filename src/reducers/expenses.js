//Expenses Reducer

const expensesReducerDefaultState = [];

export default (state = expensesReducerDefaultState, action) => {
    switch (action.type) {
        default:
            return state;
        case 'ADD_EXPENSE':
            return [
                ...state,
                action.expense
            ];
        case 'REMOVE_EXPENSE':
            return state.filter(({ id }) => id !== action.id);
        case 'EDIT_EXPENSE':
            return state.map((expense) => {
                if (expense.id === action.id) {
                    return {                // return new object
                        ...expense,         // pass all its existing properties down
                        ...action.updates   // override properties input by user
                    }
                } else {
                    return expense;
                }
            });
    }
};

