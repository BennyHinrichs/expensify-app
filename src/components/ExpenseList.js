import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses';

export const ExpenseList = (props) => (
    <div className="content-container">
        <div className="list-header">
            <div className="show-for-mobile">Expenses</div>
            <div className="show-for-large">Expense</div>
            <div className="show-for-large">Amount</div>
        </div>
        <div className="list-body">
            {props.expenses.length === 0 && 
                <span className="list-item list-item--message">
                    No expenses logged
                </span>
            }
            {
                props.expenses.map((expense) => (
                    <ExpenseListItem
                        {...expense}
                        key={expense.id}
                    />
                ))
            }
        </div>
    </div>
);

const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filters)
        // without the filter (obviously won't filter the object):
        // expenses: state.expenses,
        // filters: state.filters
    };
};

export default connect(mapStateToProps)(ExpenseList);