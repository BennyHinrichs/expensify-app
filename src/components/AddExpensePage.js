import React from 'react';
import ExpenseForm from './ExpenseForm';
import { connect } from 'react-redux';
import { addExpense } from '../actions/expenses';

export class AddExpensePage extends React.Component {
    onSubmit = (expense) => {
        this.props.addExpense(expense);
        this.props.history.push('/'); // redirect to dashboard
    };

    render() {
        return (
            <div>
                <h1>Add Expense</h1>
                <ExpenseForm 
                    path={this.props.location.pathname}
                    onSubmit={this.onSubmit}
                />
            </div>
        );
    };
}

const mapDispatchToProps = (dispatch) => ({ addExpense: (expense) => dispatch(addExpense(expense)) });

// mapDispatchToProps is the second argument of connect. if mapStateToProps is empty, must be undefined.
export default connect(undefined, mapDispatchToProps)(AddExpensePage);