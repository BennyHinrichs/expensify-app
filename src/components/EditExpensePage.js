import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { startEditExpense, startRemoveExpense } from '../actions/expenses';
import { setTimeout } from 'timers';

export class EditExpensePage extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            confirmMessage: ''
        }
    }

    onSubmit = (expense) => {
        this.props.startEditExpense(this.props.expense.id, expense);
        this.props.history.push('/');
    };

    onRemove = (e) => {
        e.preventDefault();
        if (!this.state.confirmMessage) {
            this.setState(() => ({ confirmMessage: 'Press delete again to confirm' }));
            setTimeout(() => this.setState(() => ({ confirmMessage: '' })), 3000);
            setTimeout(() => document.getElementById('message').style.color = 'white', 1);
        } else {
            this.props.startRemoveExpense({ id: this.props.expense.id });
            this.props.history.push('/');
            // this.setState(() => ({ confirmMessage: '' }));
        }
    };

    render() {
        return (
            <div>
                <div className="page-header">
                    <div className="content-container">
                        <h1 className="page-header__title">Edit Expense</h1>
                    </div>
                </div>
                <div className="content-container">
                    <ExpenseForm
                        expense={this.props.expense}
                        onSubmit={this.onSubmit}
                    />
                    <button 
                        className="button button--secondary"
                        id="button--secondary"
                        onClick={this.onRemove}
                    >Delete</button>
                    {this.state.confirmMessage && 
                        (<span className="button--secondary__message" id="message">
                            {this.state.confirmMessage} 
                        </span>
                        )
                    }
                </div>
            </div>
        );
    }
}

const mapStateToProps = (state, props) => {
    return {
        expense: state.expenses.find((expense) => expense.id === props.match.params.id),
        //description: state.description.find((expense) => expense.id === props.match.params.id)
    };
};

const mapDispatchToProps = (dispatch, props) => {
    return {
        startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
        startRemoveExpense: (data) => dispatch(startRemoveExpense(data))
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);