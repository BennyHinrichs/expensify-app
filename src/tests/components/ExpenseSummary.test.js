import React from 'react';
import { ExpenseSummary } from '../../components/ExpenseSummary';
import { shallow } from 'enzyme';
import expenses from '../fixtures/expenses';

test('should properly render summary of expenses', () => {
    const wrapper = shallow(<ExpenseSummary expenses={expenses} />);
    expect(wrapper).toMatchSnapshot();
});

test('should properly render 1 expense', () => {
    const wrapper = shallow(<ExpenseSummary expenses={[expenses[0]]} />);
    expect(wrapper).toMatchSnapshot();
});