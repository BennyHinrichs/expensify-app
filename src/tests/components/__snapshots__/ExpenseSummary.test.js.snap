// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should properly render 1 expense 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        1
      </span>
       
      expense
       totalling 
      <span>
        $
        40.00
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`should properly render summary of expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        3
      </span>
       
      expenses
       totalling 
      <span>
        $
        605.00
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
